src/foam/core/crunch/UserCapabilityJunctionRefine.js

```diff
@@ -447,6 +453,25 @@ foam.CLASS({
           (await this.sourceId$find)?.legalName;
       }
     }
+  ],
+
+  actions: [
+    {
+      name: 'reset',
+      label: 'Reset UCJ Data',
+      availablePermissions: [ 'usercapabilityjunction.action.reset' ],
+      isAvailable: (data) => !! data,
+      confirmationRequired: () => true,
+      code: async function() {
+        const ret = await this.crunchService.resetJunctionData(null, this.id);
+        if ( ret ) {
+          this.copyFrom(ret);
+          this.data = null;
+          this.notify(this.RESET_SUCCESS, '', this.LogLevel.INFO, true);
+        }
+        this.finished.pub();
```

Comment:
This doesn't need its own finished topic, can just use the one on the detailView as long as you use the context provided in the first argument of the action
```suggestion
        X.detailView?.finished.pub();
```